File: frontend/src/schemas/health.ts
import { z } from "zod";

export const dependencyStatusSchema = z.enum([
  "ok",
  "unavailable",
  "timeout",
  "unconfigured",
]);

export const livenessResponseSchema = z.object({ status: z.literal("ok") }).strict();

export const dependencyStatusesSchema = z
  .object({ postgres: dependencyStatusSchema, minio: dependencyStatusSchema })
  .strict();

export const readinessResponseSchema = z
  .object({ status: z.literal("ready"), dependencies: dependencyStatusesSchema })
  .strict();

export type LivenessResponse = z.infer<typeof livenessResponseSchema>;
export type DependencyStatuses = z.infer<typeof dependencyStatusesSchema>;
export type ReadinessResponse = z.infer<typeof readinessResponseSchema>;

File: src/invoiceops_agent/api/schemas/health.py
"""Liveness and dependency-readiness response contracts."""

from typing import Literal

from pydantic import BaseModel, ConfigDict

DependencyStatus = Literal["ok", "unavailable", "timeout", "unconfigured"]


class LivenessResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    status: Literal["ok"] = "ok"


class DependencyStatuses(BaseModel):
    model_config = ConfigDict(extra="forbid")

    postgres: DependencyStatus
    minio: DependencyStatus


class ReadinessResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    status: Literal["ready"] = "ready"
    dependencies: DependencyStatuses
